lib/sycamore/sycamore/transforms/text_extraction/ocr_models.py

```diff
@@ -28,69 +28,60 @@ def get_text(self, image: Image.Image) -> str:
         pass
 
     @abstractmethod
-    def get_boxes_and_text(self, image: Image.Image) -> List[Dict[str, Any]]:
+    def get_boxes_and_text(self, image: Image.Image) -> list[dict[str, Any]]:
         pass
 
-    @timetrace("OCR Page Extraction")
-    def extract_page(self, page: Union["Image.Image", "LTPage"]) -> List[Element]:
-        assert isinstance(page, Image.Image)
-        ocr_output = self.get_boxes_and_text(page)
-        width, height = page.size
-        texts: List[Element] = []
+    def parse_ocr_output(self, ocr_output: list[dict[str, Any]], width, height) -> list[Element]:
+        texts: list[Element] = []
         for obj in ocr_output:
-            if obj["bbox"] and not obj["bbox"].is_empty() and obj["text"] and len(obj["text"]) > 0:
+            obj_bbox = obj.get("bbox")
+            obj_text = obj.get("text")
+            if obj_bbox and not obj_bbox.is_empty() and obj_text:
                 text = Element()
                 text.type = "text"
                 text.bbox = BoundingBox(
-                    obj["bbox"].x1 / width,
-                    obj["bbox"].y1 / height,
-                    obj["bbox"].x2 / width,
-                    obj["bbox"].y2 / height,
+                    obj_bbox.x1 / width,
+                    obj_bbox.y1 / height,
+                    obj_bbox.x2 / width,
+                    obj_bbox.y2 / height,
                 )
-                text.text_representation = obj["text"]
+                text.text_representation = obj_text
                 texts.append(text)
         return texts
 
-    @timetrace("OCR Document Extraction")
+    @timetrace("OCRPageEx")
+    def extract_page(self, page: Optional[Union["PDFPage", "Image.Image"]]) -> list[Element]:
+        assert isinstance(page, Image.Image)
+        ocr_output = self.get_boxes_and_text(page)
+        width, height = page.size
+        texts: list[Element] = self.parse_ocr_output(ocr_output, width, height)
+        return texts
```

Comment:
If you just return `self.parse_ocr_output(...)` you can sidestep the unsightly annotation with no loss of type checking.